app/views/track.js

```diff
@@ -118,6 +127,42 @@ const TrackView = Mn.LayoutView.extend({
         e.stopPropagation();
     },
 
+    setClass() {
+        this.$el.removeClass();
+        this.order = [];
+
+        // Check if track is in upNext
+        let tracks = application.upNext.get('tracks');
+        tracks.each((track, index) => {
```

Comment:
Why not using `tracks.where()` to find the given track in the collectoin (or an empty array), and construct the `order` array from this last?